srt.py

```diff
@@ -213,21 +215,10 @@ def srt_timestamp_to_timedelta(ts):
     :rtype: datetime.timedelta
     """
 
-    # This function is *extremely* hot during parsing, so please keep perf in
-    # mind.
-
-    if len(ts) < TS_LEN:
-        raise ValueError(
-            "Expected timestamp length >= {}, but got {} (value: {})".format(
-                TS_LEN, len(ts), ts
-            )
-        )
-
-    # Doing this instead of splitting based on the delimiter using re.split
-    # with a compiled regex or str.split is ~15% performance improvement during
-    # parsing. We need to look from the end because the number of hours may be
-    # >2 digits.
-    hrs, mins, secs, msecs = (int(x) for x in [ts[:-10], ts[-9:-7], ts[-6:-4], ts[-3:]])
+    match = TS_REGEX.match(ts)
+    if match is None:
+        raise ValueError("Got unparseable timestamp: {}".format(ts))
+    hrs, mins, secs, msecs = map(int, match.groups())
```

Comment:
Seems fine, but we should double check perf isn't too bad here. I'm not too worried about parsing a single SRT file, more about the capacity for those who are using `srt.parse` over and over and over. This is more an action item for me once the code is updated though :-)